eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventHubClientTest.java

```diff
@@ -297,6 +344,15 @@ private static ConnectionStringProperties getCredentials(URI endpoint, String ev
         return new ConnectionStringProperties(connectionString);
     }
 
+    private Mono<Void> pushEventsToPartition(final EventHubClient client, final EventHubProducerOptions senderOptions, final int numberOfClients) {
+        final Flux<EventData> events = Flux.range(0, numberOfClients).map(number -> {
+            final EventData data = new EventData("testString".getBytes(UTF_8));
+            return data;
+        });
+        sender = client.createProducer(senderOptions);
```

Comment:
I dislike this method because it's doing two very different things. It's
1. creating a resource (producer)
1. sending data.

who owns the clean up of this producer when the send completes?